### PR TITLE
fix: Add backoff to create pod operation

### PR DESCRIPTION
**Pull request by copierrj**
Creating a pod for a workflow step is currently not retried on failure. This results in failing workflows due to intermittent k8s api availability issues. Because these availability issues are often difficult to eliminate entirely we propose to retry pod creation.

Checklist:

* [x] My organization is added to [USERS.md](https://github.com/argoproj/argo-workflows/blob/master/USERS.md).

### CODE REVIEW

File: workflow/controller/workflowpod.go [hunk 1]
```diff
@@ -414,17 +416,22 @@ func (woc *wfOperationCtx) createWorkflowPod(ctx context.Context, nodeName strin
...
 
-	created, err := woc.controller.kubeclientset.CoreV1().Pods(woc.wf.ObjectMeta.Namespace).Create(ctx, pod, metav1.CreateOptions{})
-	if err != nil {
-		if apierr.IsAlreadyExists(err) {
-			// workflow pod names are deterministic. We can get here if the
-			// controller fails to persist the workflow after creating the pod.
-			woc.log.Infof("Failed pod %s (%s) creation: already exists", nodeName, nodeID)
-			return created, nil
-		}
-		if errorsutil.IsTransientErr(err) {
```
Comment (terrytangyuan) [inline workflow/controller/workflowpod.go:425]: It should already be retried in transient errors. The controller would requeue this when transient error happens.

Comment (terrytangyuan) [reply workflow/controller/workflowpod.go:425]: Ref: https://github.com/argoproj/argo-workflows/blob/master/workflow/controller/operator.go#L2234-L2242

Comment (sarabala1979) [reply workflow/controller/workflowpod.go:425]: Can you provide what kind of error it is failing? if it is in transient error which missed in requeueIfTransientErr, you can add it

Comment (copierrj) [reply workflow/controller/workflowpod.go:425]: We've observed two errors (both of type *url.Error) with the following messages:
Post https://192.168.254.1:443/api/v1/namespaces/delivery/pods: read tcp 10.244.56.73:35718->192.168.254.1:443: read: connection reset by peer"

Comment (sarabala1979) [reply workflow/controller/workflowpod.go:425]: It looks like your k8s API server is unavailable or high load. But in normal scenarios, these errors are not considered transient errors. I would recommend adding retrystrategy on those steps.  `Backoff` will create an extra load on API server.

**Review by alexec — COMMENTED** [review]
I don't think we need this. Failures will be automatically re-queued already.

**Comment by copierrj** [comment]
> I don't think we need this. Failures will be automatically re-queued already.

You are right. Although we probably have to set TRANSIENT_ERROR_PATTERN.

**Comment by terrytangyuan** [comment]
> > I don't think we need this. Failures will be automatically re-queued already.
> 
> You are right. Although we probably have to set TRANSIENT_ERROR_PATTERN.

Yes also note you can customize the backoff behavior via environment variables as well. Search for `RETRY_` in : https://argoproj.github.io/argo-workflows/environment-variables/

**Comment by simster7** [comment]
We've discussed with the team and we don't believe this is something that needs fixing

**Comment by tooptoop4** [comment]
where @copierrj

**Pull request closed without merging**